src/preprocessing/preprocessing_pass_context.h

```diff
@@ -66,7 +66,7 @@ class PreprocessingPassContext
     d_resourceManager->spendResource(r);
   }
 
-  const LogicInfo& getLogicInfo() { return d_smt->d_logic; }
+  LogicInfo getLogicInfo() { return d_smt->getLogicInfo(); }
```

Comment:
Please keep the `const&` here if possible, and instead make `SmtEngine::getLogicInfo` also return a `const&`.